lib/daos/subjectsDao.js

```diff
@@ -0,0 +1,64 @@
+import { subjects } from 'models';
+import get from 'lodash/get';
+import sequelize from 'sequelize';
+import { badRequest, notFound } from 'utils/responseInterceptors';
+import { transformDbArrayResponseToRawResponse } from 'utils/transformerUtils';
+
+const attributes = ['id', 'name'];
+export const findOneSubject = id =>
+    subjects.findOne({ where: { id }, attributes });
+
+export const findAllSubjects = async (where, page, limit, include) => {
+    const totalCount = await subjects.count({ where });
+    const results = await subjects.findAll({
+        where,
+        attributes,
+        order: sequelize.literal('id ASC'),
+        limit,
+        offset: (page - 1) * limit,
+        include
+    });
+    if (get(results, 'length')) {
+        const allSubjects = transformDbArrayResponseToRawResponse(results).map(
+            subject => subject
+        );
+        return {
+            results: allSubjects,
+            totalCount
+        };
+    }
+    return notFound('No subjects found');
+};
+
+export const updateOneSubject = async (id, subject) => {
```

Comment:
```
export const updateOneSubject = async (id, updatedSubject) => {
    let subject = await findOneSubject(id);
    if (subject) {
       subject = await subject.update(updatedSubject);
       return {id: subject.id, name: subject.name}
    }
    return badRequest('Did not update anything, please check the id');
}
```

Reply:
fixed the updateOneSubject and deleteOneSubject functions